src/main/java/net/fabricmc/loader/impl/launch/MappingConfiguration.java

```diff
@@ -159,6 +162,19 @@ private void initializeMappings() {
 					throw new UnsupportedOperationException("Unsupported mapping format: " + format);
 				}
 
+				if (!mappings.getDstNamespaces().contains("official")) {
+					MemoryMappingTree completedTree = new MemoryMappingTree();
+
+					Map<String, String> completableToComplete = new HashMap<>();
+					completableToComplete.put("clientOfficial", "intermediary");
+					completableToComplete.put("serverOfficial", "intermediary");
+
+					MappingNsCompleter completer = new MappingNsCompleter(completedTree, completableToComplete);
+					mappings.accept(completer);
+
+					mappings = completedTree;
```

Comment:
Why is this needed? We dont do it for for the regular mappings.

Reply:
This is done because intermediary is the only complete namespace. The other two have 'holes' which is problematic when providing them to tiny remapper.

Reply:
Whats the issue, ideally we should be handling holes in mappings instead of just patch fixing it with MappingNsCompleter. Its something I have been gulity of doing many times, would just be nice to fix the root issue rarther than hiding it.

Reply:
It's mainly because the code which converts mio's MappingTree to tr's IMappingProvider assume every entry exists in both source and target namespaces. 
Which is not the case in merged intermediaries for entries that exists only on one side, the side missing the name will have it being empty. In this case mio returns null, which tr can't handle.

The two known way to fix this issue currently are completing the mappings or replacing null values in TinyRemapperMappingsHelper with the name used in target namespace.
Loom pr used the first way.
Cursed Legacy and Babric have been using the second one since they exist before mio was created.

Reply:
The 'holes' are due to members that are one side only, so ideally those intermediary entries should be dropped on the other side.

Reply:
@thecatcore Can you elaborate / give an example? When I added transitioned TR to use MIO under the hood, I explicitly added null checks to both the source and destination sides

Reply:
hm, it seems like TR doesn't have a builtin method to create IMappingProvider directly from MIO MappingTree or MappingVisitor. [TR's TinyUtils](https://github.com/FabricMC/tiny-remapper/blob/master/src/main/java/net/fabricmc/tinyremapper/TinyUtils.java)

Reply:
Yeah, currently we only support reading from disk. How would a Tiny v2 file look like such that it would cause the issue you described? The null checks I mentioned should be enough.

Reply:
https://github.com/FabricMC/fabric-loom/blob/dev/1.6/src/test/resources/mappings/1.2.5-intermediary.tiny.zip

Reply:
I've ran some tests, and Tiny Remapper handles everything just fine, both server and client, so there's no need for reordering the mappings yourself

Reply:
Done: https://github.com/FabricMC/tiny-remapper/pull/128